app/controllers/organizations_controller.rb

```diff
@@ -146,25 +137,52 @@ def set_users_github_organizations
     end
   end
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
   def set_filter_options
     @sort_modes = Organization.sort_modes
+    @view_modes = Organization.view_modes
+
+    @current_sort_mode = if @sort_modes.dig(params[:sort_by])
+                           params[:sort_by]
+                         else
+                           @sort_modes.keys.first
+                         end
+
+    @current_view_mode = if @view_modes.dig(params[:view])
+                           params[:view]
+                         else
+                           @view_modes.keys.first
+                         end
 
-    @current_sort_mode = params[:sort_by] || @sort_modes.keys.first
     @query = params[:query]
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
 
-    @sort_modes_links = @sort_modes.keys.map do |mode|
-      search_organizations_path(
-        sort_by: mode,
-        query: @query
-      )
-    end
+  # rubocop:disable Metrics/MethodLength
+  def set_filtered_organizations
+    scope = current_user.organizations.includes(:assignments, :group_assignments).filter_by_search(@query)
+
+    scope = case @current_view_mode
+            when "Archived" then scope.archived
+            when "Active" then scope.not_archived
+            else scope
+            end
+
+    @organizations = scope
+      .order_by_sort_mode(@current_sort_mode)
+      .order(:id)
+      .page(params[:page])
+      .per(12)
   end
+  # rubocop:enable Metrics/MethodLength
 
   # Check if the current user has any organizations with admin privilege,
   # if so add the user to the corresponding classroom automatically.
   def add_current_user_to_organizations
     @users_github_organizations.each do |github_org|
-      user_classrooms = Organization.where(github_id: github_org[:github_id])
+      user_classrooms = Organization.includes(:users).where(github_id: github_org[:github_id])
```

Comment:
I added this to get one of the eager load checks to stop yelling at me every page load.